packages/lodestar/src/db/api/beacon/repositories/attestation.ts

```diff
@@ -10,10 +12,31 @@ import {IDatabaseController, Bucket, Repository} from "@chainsafe/lodestar-db";
  * Removed when included on chain or old
  */
 export class AttestationRepository extends Repository<Uint8Array, phase0.Attestation> {
+  // For each slot, index attestation roots by attestation data root hex
+  // do batch remove per finalized checkpoint
+  private dataRootIndex = new Map<Slot, Map<string, string[]>>();
+
   constructor(config: IBeaconConfig, db: IDatabaseController<Buffer, Buffer>) {
     super(config, db, Bucket.phase0_attestation, config.types.phase0.Attestation);
   }
 
+  async put(key: Uint8Array, value: phase0.Attestation): Promise<void> {
+    const attestationDataRoot = this.config.types.phase0.AttestationData.hashTreeRoot(value.data);
+    const slot = value.data.slot;
+    const dataRootHex = toHexString(attestationDataRoot);
+    const rootHex = toHexString(key);
+    if (!this.dataRootIndex.get(slot)) {
```

Comment:
You might structure this part like:
```ts
let slotIndex = this.dataRootIndex.get(slot);
if (!slotIndex) {
  slotIndex = new Map();
  this.dataRootIndex.set(slot, slotIndex);
}
slotIndex.get(...)....
```
to avoid repeated `this.dataRootIndex.get(slot)?` in the rest.